src/command_modules/azure-cli-storage/tests/test_storage_blob_scenarios.py

```diff
@@ -70,55 +65,51 @@ def test_storage_blob_upload_64mb_file(self, resource_group, storage_account):
-        account_key = self.get_account_key(group, account)
+        account_info = self.get_account_info(group, account)
 
-        self.set_env('AZURE_STORAGE_ACCOUNT', account)
-        self.set_env('AZURE_STORAGE_KEY', account_key)
+        container = self.create_container(account_info)
 
-        self.cmd('storage container create -n {}'.format(container))
+        self.storage_cmd('storage blob exists -n {} -c {}', account_info, blob_name, container)
+        self.storage_cmd('storage blob upload -c {} -f {} -n {} --type {}', account_info,
+                         container, local_file, blob_name, blob_type)
+        self.storage_cmd('storage blob exists -n {} -c {}', account_info, blob_name, container) \
+            .assert_with_checks(JMESPathCheck('exists', True))
 
-        self.cmd('storage blob exists -n {} -c {}'.format(blob_name, container),
-                 checks=JMESPathCheck('exists', False))
-
-        self.cmd('storage blob upload -c {} -f {} -n {} --type {}'
-                 .format(container, local_file, blob_name, blob_type))
-
-        self.cmd('storage blob exists -n {} -c {}'.format(blob_name, container),
-                 checks=JMESPathCheck('exists', True))
-
-        self.cmd('storage blob show -n {} -c {}'.format(blob_name, container),
-                 checks=JMESPathCheck('name', blob_name))  # TODO: more checks
+        self.storage_cmd('storage blob show -n {} -c {}', account_info, blob_name, container) \
+            .assert_with_checks(JMESPathCheck('name', blob_name))  # TODO: more checks
 
         expiry = (datetime.utcnow() + timedelta(hours=1)).strftime('%Y-%m-%dT%H:%MZ')
-        sas = self.cmd('storage blob generate-sas -n {} -c {} --expiry {} '
-                       '--permissions r --https-only'.format(blob_name, container, expiry)).output
-        assert dict(pair.split('=') for pair in sas.split('&'))  # TODO: more checks
+        sas = self.storage_cmd('storage blob generate-sas -n {} -c {} --expiry {} --permissions '
+                               'r --https-only', account_info, blob_name, container, expiry).output
+        self.assertTrue(sas)
+        self.assertIn('sig', sas)
 
-        self.cmd('storage blob update -n {} -c {} --content-type application/test-content'
-                 .format(blob_name, container))
+        self.storage_cmd('storage blob update -n {} -c {} --content-type application/test-content',
+                         account_info, blob_name, container)
 
-        self.cmd('storage blob show -n {} -c {}'.format(blob_name, container), checks=[
-            JMESPathCheck('properties.contentSettings.contentType', 'application/test-content'),
-            JMESPathCheck('properties.contentLength', file_size_kb * 1024)])
+        self.storage_cmd('storage blob show -n {} -c {}', account_info, blob_name, container) \
+            .assert_with_checks([JMESPathCheck('properties.contentSettings.contentType', 'application/test-content'),
```

Comment:
I would request we either make self.cmd support .assert_with_checks as well or that we make self.storage_cmd work with the checks kwarg... or remove it altogether.  I don't see what it does that isn't accomplished by simple text replacement of the connection parameters.  

My concern is having fragmented methods of accomplishing the same thing making it difficult to recommend to third parties the right way to do things.